### PR TITLE
chore: update OTel deps

### LINKED ISSUE #207: Reference renamed packages in OTel
Many packages are going through a rename to unify the naming schema across them. That will affect our distro as well.

See https://github.com/open-telemetry/opentelemetry-js/pull/2340

### PR DESCRIPTION
**Pull request by Rauno56**
# Description

Update deps to switch over to [the ones that were renamed](https://github.com/open-telemetry/opentelemetry-js/pull/2340).

fixes #207

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Internal change (a change which is not visible to the package consumers)
- [ ] Documentation change or requires a documentation update

# How Has This Been Tested?

- [ ] Tested manually
- [ ] Added automated tests

# Checklist:

- [ ] Unit tests have been added/updated
- [ ] Documentation has been updated
- [ ] Change file has been generated (`npm run change:new`)
- [ ] Delete this branch (after the PR is merged)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,9 +55,9 @@
     "@commitlint/cli": "^13.1.0",
     "@commitlint/config-conventional": "^13.1.0",
     "@octokit/core": "^3.5.1",
-    "@opentelemetry/instrumentation-bunyan": "~0.24.0",
-    "@opentelemetry/instrumentation-http": "~0.23.0",
-    "@opentelemetry/instrumentation-pino": "~0.23.0",
+    "@opentelemetry/instrumentation-bunyan": "^0.24.0",
+    "@opentelemetry/instrumentation-http": "^0.24.0",
+    "@opentelemetry/instrumentation-pino": "^0.24.0",
```
Comment (owais) [inline package.json:60]: Why did we change `~` to `^` ?

Comment (owais) [reply package.json:60]: I highly suggest we use `~` for packages that are pre 1.0 and use `^` only for stable packages

Comment (Rauno56) [reply package.json]: You're right. Changed those.

**Review by owais — APPROVED** [review]

File: change/@splunk-otel-ba9ac324-9f97-419b-890e-aa3e6b8bd70a.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (jtmal-signalfx) [inline change/@splunk-otel-ba9ac324-9f97-419b-890e-aa3e6b8bd70a.json:2]: if OTel was bumped by `minor`, then shouldn't we bump by minor too?

Comment (jtmal-signalfx) [reply change/@splunk-otel-ba9ac324-9f97-419b-890e-aa3e6b8bd70a.json:2]: Assuming that you reviewed changes to the API and the SDK and concluded that the minor bump is not needed - then no!

**Review by jtmal-signalfx — APPROVED** [review]

**Pull request merged**